CRM/Admin/Form/Options.php

```diff
@@ -431,6 +433,7 @@ public function postProcess() {
       }
 
       $groupParams = array('name' => ($this->_gName));
+      // Option values get stored here. Might be some problem with addOptionValue.
```

Comment:
Again. addOptionValue performs some data validation and then calls CRM_Core_BAO_OptionValue::add which again performs some data validation and creates a new DAO object.